drivers/sensor/pms7003/pms7003.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* sensor pms7003.c - Driver for plantower PMS7003 sensor
+ * PMS7003 product: http://www.plantower.com/en/content/?110.html
+ * PMS7003 spec: http://aqicn.org/air/view/sensor/spec/pms7003.pdf
+ */
+
+#include <errno.h>
+
+#include <arch/cpu.h>
+#include <init.h>
+#include <kernel.h>
+#include <sensor.h>
+#include <stdlib.h>
+#include <string.h>
+#include <uart.h>
+
+#define SYS_LOG_DOMAIN "PMS7003"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SENSOR_LEVEL
+#include <logging/sys_log.h>
+
+/* wait serial output with 1000ms timeout */
+#define CONFIG_PMS7003_SERIAL_TIMEOUT 1000
+
+struct pms7003_data {
+	struct device *uart_dev;
+	u16_t pm1;
+	u16_t pm25;
+	u16_t pm10;
+};
+
+/**
+ * @brief wait for an array data from uart device with a timeout
+ *
+ * @param dev the uart device
+ * @param data the data array to be matched
+ * @param len the data array len
+ * @param timeout the timeout in milliseconds
+ * @return 0 if success; -ETIME if timeout
+ */
+int uart_wait_for(struct device *dev, uint8_t *data, int len, int timeout)
+{
+	int matched_size = 0;
+	s64_t timeout_time = k_uptime_get() + K_MSEC(timeout);
+
+	while (1) {
+		unsigned char c;
+		if (k_uptime_get() > timeout_time) {
+			return -ETIME;
+		}
+		if (uart_poll_in(dev, &c) == 0) {
+			if (c == data[matched_size]) {
+				matched_size++;
+				if (matched_size == len) {
+					break;
+				}
+			} else if (c == data[0]) {
+				matched_size = 1;
+			} else {
+				matched_size = 0;
+			}
+		}
+	}
+	return 0;
+}
+
+/**
+ * @brief read bytes from uart
+ *
+ * @param data the data buffer
+ * @param len the data len
+ * @param timeout the timeout in milliseconds
+ * @return 0 if success; -ETIME if timeout
+ */
+int uart_read_bytes(struct device *dev, uint8_t *data, int len, int timeout)
+{
+	int read_size = 0;
+	s64_t timeout_time = k_uptime_get() + K_MSEC(timeout);
+
+	while (1) {
+		unsigned char c;
+		if (k_uptime_get() > timeout_time) {
```

Comment:
same here, well you get the point. Aerate your code a bit.

You can avoid breaking a line before an if/for/while when the previous statement is a variable assignment that is tested in the condition afterwards.